src/librustc_ast_lowering/lib.rs

```diff
@@ -648,14 +668,14 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         })
     }
 
-    fn next_id(&mut self) -> hir::HirId {
+    fn next_id(&mut self, span: Span) -> hir::HirId {
```

Comment:
I am trying to decide whether it is not intuitive for `fn next_id` to take a `span` as a parameter.

I think I understand the *intent* you have here: Every call to `next_id` is matched up with some portion of the HIR that is being generated via the lowering of the AST, and each such portion of the HIR is meant to always have a span associated with it. So in that sense, it makes sense to pass along the `span`.

But at the same time, when I *read* the invocations written as `self.next_id(span)`, my own intuitive reading is "this is generating an `id` *from* the span somehow" (which is certainly not what is happening).

I'll keep musing on this. In the meantime, I would suggest you add a doc-comment to `fn next_id` saying what the `span` parameter is. (I also played with the idea of suggesting renaming the method, but I wasn't really happy with any of the alternatives I came up with...)